mono/metadata/loader-internals.h

```diff
@@ -89,6 +89,8 @@ struct _MonoMemoryManager {
 	gboolean is_generic;
 	// Whether the MemoryManager is in the process of being freed
 	gboolean freeing;
+	// Whenever this is owned by a dynamic method
+	gboolean dynamic_method;
```

Comment:
I think you're adding a third kind of memory manager here.  So maybe it's better to do something like: 

```c
enum  { MEM_MGR_SINGLE, MEM_MGR_GENERIC, MEM_MGR_DYNAMIC_METHOD } MonoMemoryManagerKind;

struct _MonoMemoryManager {
  MonoMemoryManagerKind kind;
   ...
};

struct _MonoMemoryManagerDynamicMethod {
   MonoMemoryManager  memory_manager;
};
```